### PR TITLE
Hide all other series when alt-clicking in the legend

### LINKED ISSUE #4548: egui_plot legend: add shortcut to hide all series but this one
This could be achieved e.g. by alt-clicking on a series name.

### PR DESCRIPTION
**Pull request by abey79**
https://github.com/emilk/egui/assets/49431240/75d32f53-4c1c-4713-b25e-e1787e465a48

* Closes #4548

### CODE REVIEW

File: crates/egui_plot/src/legend.rs [hunk 1]
```diff
@@ -267,14 +267,43 @@ impl Widget for &mut LegendWidget {
...
-                        entries
+                        let mut focus_on_item = None;
+
+                        let response_union = entries
                             .iter_mut()
-                            .map(|(name, entry)| entry.ui(ui, name.clone(), &config.text_style))
+                            .map(|(name, entry)| {
+                                let response = entry.ui(ui, name.clone(), &config.text_style);
+
+                                if response.clicked() && ui.input(|r| r.modifiers.alt) {
```
Comment (emilk) [inline crates/egui_plot/src/legend.rs:277]: Where is the normal clicking handled? We shouldn't spread out the different click-handling to different places

Comment (abey79) [reply crates/egui_plot/src/legend.rs:277]: good call. i'll refactor some stuff.

**Review by emilk — APPROVED** [review]

**Comment by emilk** [comment]
Where can we document this feature? Would a tooltip when hovering the legend be too annoying?

**Comment by abey79** [comment]
> Where can we document this feature? Would a tooltip when hovering the legend be too annoying?

I've pondered about this and decided against the tooltip—too annoying and difficult to override for the host application. I guess I can add a feature flag for this in `Legend`, such that the functionality appears in the docs. I'd argue that the host application is responsible for controlling what the user can do and provide help about it (e.g. also double click to reset, etc.).

For rerun, I have it in my todos to add something in the space view help.

**Pull request merged**